src/components/Shipping/Shipping.tsx

```diff
@@ -1,28 +1,25 @@
-import React, {useState} from 'react'
+import React, {useState, useContext } from 'react'
 import './Shipping.scss'
 import ShippingHome from "../../components/ShippingHome/ShippingHome"
 import ShippingStore from "../../components/ShippingStore/ShippingStore"
 import ShippingData from "../../components/ShippingData/ShippingData"
+import {FormComponentContext} from '../../contexts/FormComponentContext'
+import {CartComponentContext, CartContextProvider} from '../../contexts/CartComponentContext'
 
-interface IProps {
-    componentShipping: boolean;
-    openComponentShipping: any;
-    setTax: any;
-    setCheckValue: any;
-    checkValue: number;
-    shipValue: number;
-    openComponentPayment: any;
-    openEditBtn: any;
-    editBtn: any;
-}
+const Shipping:React.FC= () => {
+
+    const formContext:any = useContext(FormComponentContext)
+    const {formComponents, dispatch} = formContext
+    const {componentShipping, editShipping, storeShipping, homeShipping} = formComponents
 
-const Shipping:React.FC<IProps> = ({componentShipping, openComponentShipping, setTax, setCheckValue, checkValue, shipValue, openComponentPayment, openEditBtn, editBtn}) => {
+    // const cartContext:any = useContext(CartComponentContext)
+    // const {dispatch} = cartContext
 
     const submitBtn = () => {
-        setTax(true)
-        openComponentShipping(!componentShipping)
-        openComponentPayment(true)
-        openEditBtn({...editBtn, shipping: true})
+        dispatch({type: "COMPONENT", payload: {tax: true}})
+        dispatch({type: "COMPONENT", payload: {componentShipping: !componentShipping}})
+        dispatch({type: "COMPONENT", payload: {componentPayment: true}})
+        dispatch({type: "COMPONENT", payload: {editShipping: true}})
```

Comment:
Why do you need 4 dispatches? Can it be done with one action that has complex payload?